src/mlpack/core/tree/hollow_ball_bound.hpp

```diff
@@ -35,6 +36,8 @@ class HollowBallBound
   math::RangeType<ElemType> radii;
   //! The center of the ball bound.
   arma::Col<ElemType> center;
+  //! The center of the hollow.
+  arma::Col<ElemType> hollowCenter;
```

Comment:
I thought to myself, 'surely, holding two centers, which can double the number of distance computations or more for a single score, can't be as fast as only holding one, even if the bounds end up being a little looser'.  But after I tried implementing it, and removing the `hollowCenter` member, the runtimes were always about 10% slower.  So it seems that `hollowCenter` is the right way to go here.